### ISSUE TITLE
ignoreInvalidPercentEncoding encodes %s of correctly encoded URLs

### ISSUE THREAD

**Issue by benjben** [lemonlabsuk/scala-uri#30]
It seems that `ignoreInvalidPercentEncoding` encodes the`%`s of correctly encoded URLs.

Code Example:
```scala
import io.lemonlabs.uri.Url
import io.lemonlabs.uri.config.UriConfig
import io.lemonlabs.uri.decoding.PercentDecoder
​
implicit val c = UriConfig(
  decoder = PercentDecoder(ignoreInvalidPercentEncoding = true)
)
val url = Url.parse("example.com/path%20with%20space")
url.toString // "example.com/path%2520with%2520space"
```
`"%20"` becomes `"%2520"`

**Comment by theon** [comment]
Thanks for raising this! I've released version `1.4.2` with a fix for this.

**Comment by benjben** [comment]
Awesome, thank you for fixing it so quickly @theon!
